toolchain/check/handle_class.cpp

```diff
@@ -666,6 +666,62 @@ static auto AddStructTypeFields(
   return fields_id;
 }
 
+// Builds and returns a vtable for the current class. Assumes that the virtual
+// functions for the class are listed as the top element of the `vtable_stack`.
+static auto BuildVtable(Context& context, Parse::NodeId node_id,
+                        SemIR::InstId base_vtable_id) -> SemIR::InstId {
+  llvm::SmallVector<SemIR::InstId> vtable;
+  if (base_vtable_id.has_value()) {
+    LoadImportRef(context, base_vtable_id);
+    base_vtable_id =
+        context.constant_values().GetConstantInstId(base_vtable_id);
+    if (base_vtable_id == SemIR::ErrorInst::SingletonInstId) {
+      return SemIR::ErrorInst::SingletonInstId;
+    }
+    auto base_vtable_inst_block =
+        context.inst_blocks().Get(context.insts()
+                                      .GetAs<SemIR::Vtable>(base_vtable_id)
```

Comment:
Maybe declare a new name? Overwriting the value makes me assume that the original value would be used in some way, which doesn't seem to be the case here.

```suggestion
    auto canonical_vtable_id =
        context.constant_values().GetConstantInstId(base_vtable_id);
    if (canonical_vtable_id == SemIR::ErrorInst::SingletonInstId) {
      return SemIR::ErrorInst::SingletonInstId;
    }
    auto base_vtable_inst_block =
        context.inst_blocks().Get(context.insts()
                                      .GetAs<SemIR::Vtable>(canonical_vtable_id)
```